src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs

```diff
@@ -315,157 +120,66 @@ public async Task<string> GetGitCommitAsync(string? repoPath = null, Cancellatio
-                        if (oldWorkTree != null)
-                        {
-                            log.LogDebug($"resetting {subRepoPath} worktree to {oldWorkTree.Value}");
-                            subRepo.Config.Set("core.worktree", oldWorkTree.Value);
-                        }
-                        else
-                        {
-                            log.LogDebug($"leaving {subRepoPath} worktree as default");
-                        }
-                        log.LogDebug($"Done resetting {subRepoPath}, checking submodules");
-                        CleanRepoAndSubmodules(subRepo, log);
-                    }
-                }
+        var result = await _processManager.ExecuteGit(repoPath, args);
+        result.ThrowIfFailed($"Failed to find commit {gitRef} in {repoPath}");
 
-                if (File.Exists(subRepoGitFilePath))
-                {
-                    log.LogDebug($"Deleting {subRepoGitFilePath} to orphan submodule {sub.Name}");
-                    File.Delete(subRepoGitFilePath);
-                }
-                else
-                {
-                    log.LogDebug($"{sub.Name} doesn't have a .gitdir redirect at {subRepoGitFilePath}, skipping delete");
-                }
-            }
-        }
+        return result.StandardOutput.Trim();
     }
 
     /// <summary>
-    ///     Add a remote to a local repo if does not already exist, and attempt to fetch commits.
+    ///     Add a remote to a local repo if does not already exist.
     /// </summary>
-    /// <param name="repoDir">Path to a git repository</param>
+    /// <param name="repoPath">Path to a git repository</param>
     /// <param name="repoUrl">URL of the remote to add</param>
-    /// <param name="skipFetch">Skip fetching remote changes</param>
     /// <returns>Name of the remote</returns>
-    public string AddRemoteIfMissing(string repoDir, string repoUrl, bool skipFetch = false)
+    public async Task<string> AddRemoteIfMissingAsync(string repoPath, string repoUrl, CancellationToken cancellationToken = default)
     {
-        using var repo = new Repository(repoDir);
-        return AddRemoteIfMissing(repo, repoUrl, skipFetch);
-    }
+        var result = await _processManager.ExecuteGit(repoPath, new[] { "remote", "-v" }, cancellationToken: cancellationToken);
+        result.ThrowIfFailed($"Failed to get remotes for {repoPath}");
 
-    private string AddRemoteIfMissing(Repository repo, string repoUrl, bool skipFetch = false)
-    {
-        var remote = repo.Network.Remotes.FirstOrDefault(r => r.Url.Equals(repoUrl, StringComparison.InvariantCultureIgnoreCase));
-        string remoteName;
+        string? remoteName = null;
 
-        if (remote is not null)
+        foreach (var line in result.StandardOutput.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries))
```

Comment:
maybe using a Regex here is a bit easier here?
something like 
`"(\S+)\s+${repoUrl.Replace("/", "\/)}"`

Reply:
I think both of them are pretty readable and easy to understand. I guess regex on it's own, without any examples is a bit harder